pymeasure/instruments/texio/texioPSW360L30.py

```diff
@@ -0,0 +1,61 @@
+from pymeasure.instruments import Instrument
+from pymeasure.instruments.keithley.keithley2260B import Keithley2260B
+
+import time
+import logging
+
+log = logging.getLogger(__name__)
+log.addHandler(logging.NullHandler())
+
+
+class TexioPSW360L30(Keithley2260B):
+    r""" Represents the TEXIO PSW-360L30 Power Supply (minimal implementation)
+    and provides a high-level interface for interacting with the instrument.
+
+    For a connection through tcpip, the device only accepts
+    connections at port 2268, which cannot be configured otherwise.
+    example connection string: 'TCPIP::xxx.xxx.xxx.xxx::2268::SOCKET'
+
+    For a connection through USB on Linux, the kernel is going to create
+    a /dev/ttyACMX device automatically. The serial connection properties are
+    fixed at 9600–8-N-1.
+
+    The read termination for this interface is Line-Feed \n.
+
+    This driver inherits from the Keithley2260B one. All instructions
+    implemented in the Keithley 2260B driver are also available for the
+    TEXIO PSW-360L30 power supply.
+
+    The only addition is the "output" property that is just an alias for
+    the "enabled" property of the Keithley 2260B. Calling the output switch
+    "enabled" is confusing because it is not clear if the whole device is
+    enabled/disable or only the output.
+
+    .. code-block:: python
+
+        source = TexioPSW360L30("TCPIP::xxx.xxx.xxx.xxx::2268::SOCKET")
+        source.voltage = 1
+        print(source.voltage)
+        print(source.current)
+        print(source.power)
+        print(source.applied)
+    """
+
+    def __init__(self, adapter, **kwargs):
+        Instrument.__init__(self,
+                            adapter=adapter,
+                            name="TEXIO PSW-360L30 Power Supply",
+                            read_termination="\n",
+                            **kwargs
+                            )
```

Comment:
If you override a method and call the overriden method, use `super()` to access that one:
```suggestion
        super().__init__(self,
                         adapter=adapter,
                         name="TEXIO PSW-360L30 Power Supply",
                         read_termination="\n",
                         **kwargs
                         )
```

Reply:
That is not going to work because both `Keithley2260B.__init__` and `TexioPSW360L30.__init__` whould try to call `Instrument.__init__` with different arguments due to line 35 of keythley2260B: `**kwargs`

Reply:
The problem is `name`:
super().__init__ calls the Keithley init, which puts "name" into the kwargs (as it does not know a "name" parameter), then it calls Instrument.init with the name argument twice (once from Keithley init code and once from the kwargs), which raises an Exception.
There are a few options to circumvent this problem:

- Add `kwargs.setdefault('name', "Keithley 2260B DC Power Supply")` in Keithley's init before it calls super and remove the `name=Keithley` part from the super().__init parameters. That way the instrument is called Keithley unless you give it a name via kwarg (as you do it here). (My recommendation)
- call super().__init__ in the Texio init, but without a name parameter. Set `self.name = "Texio...."` after calling super().__init__
- You could call `super(Keithley2260B, self).__init__(`, because that way you get the parent of the Keithley class.

Reply:
I tried to follow your recommendation. The problem is not only the name attribute but also the positional arguments, like `adapter`.

Reply:
> You could call super(Keithley2260B, self).__init__(, because that way you get the parent of the Keithley class.

In general:  Don't try to manually override the Method Resolution Order like this. If there's something wonky here, we have to fix it properly so the MRO is set up properly, otherwise, we could miss e.g. some needed init code.

@LastStarDust I don't see off the cuff what the problem with positional arguments would be. What erroneous behaviour did you observe?

Reply:
I was trying to pass `adapter` as a keyword argument but it did not work because of multiple definitions.
If I pass it as a positional argument it does work (see the latest commit).